acme/provider.go

```diff
@@ -0,0 +1,7 @@
+package acme
+
+// ChallengeProvider makes the result of a challenge available to be solved
+type ChallengeProvider interface {
+	Present(domain, token, keyAuth string) error
+	CleanUp(domain string) error
```

Comment:
cool. i'll update shortly.

no worries i maintain enough open source projects that i know the routine from both sides =)